api/graylog.go

```diff
@@ -130,9 +129,24 @@ func UpdateRegistration(httpClient *http.Client, context *context.Ctx, status *g
 		context.UserConfig.SendStatus = false
 	} else if resp != nil && resp.StatusCode != 202 {
 		log.Error("[UpdateRegistration] Bad response from Graylog server: ", resp.Status)
-	} else if err != io.EOF { // we except EOF
+	} else if err != nil {
 		log.Error("[UpdateRegistration] Failed to report collector status to server: ", err)
 	}
+
+	// Update configuration based on server response
+	if respBody != nil {
+		// API query interval
+		if context.UserConfig.UpdateInterval != respBody.Configuration.UpdateInterval &&
+			respBody.Configuration.UpdateInterval > 0 {
+			log.Infof("[ConfigurationUpdate] update_interval: %ds", respBody.Configuration.UpdateInterval)
+			context.UserConfig.UpdateInterval = respBody.Configuration.UpdateInterval
```

Comment:
In this simple case it's safe I would say. `UpdateInterval` is initilized during startup by the config file. In the code we just update the value and dont rely on the exact order of the update. The worst case is that the old value will be used for a single next period.